torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -413,7 +413,7 @@ ExprHandle TensorExprKernel::constant(const torch::jit::Value* v) {
 
 ExprHandle promoteIntegerToFloat(const ExprHandle& e) {
   auto scalarType = static_cast<c10::ScalarType>(e.dtype().scalar_type());
-  if (!c10::isIntegralType(scalarType)) {
+  if (!c10::isIntegralType(scalarType, /* includeBool = */ false)) {
```

Comment:
This is going to conflict with : https://github.com/pytorch/pytorch/pull/47403/ where i set it to `includeBool` to true (bc in eager these ops also promote bools)